### PR TITLE
Closes #6301: Purging cache from post/page editor redirects to posts/pages list when using Classic

### LINKED ISSUE #6301: Purging cache from post/page editor redirects to posts/pages list when using Classic Editor
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**
We have three separate reports, ([one](https://secure.helpscout.net/conversation/2427287687/456004/), [two](https://secure.helpscout.net/conversation/2431551480/457262?folderId=3572910), [three](https://secure.helpscout.net/conversation/2438505643/460986?folderId=273761)) that in WPR 13.15.4 with Classic Editor on post/page edit screen, when you click on Clear cache in the admin bar, they get redirected to the list of posts.
Version 3.15.3 and earlier don't seem to be having this issue, but I can reproduce it with 3.15.4 & 3.15.5.
In version 13.15.3 when performed the same action, the post edit screen would reload, and you don't get redirected to the list of posts.

**To Reproduce**
1. Install [Classic Editor](https://wordpress.org/plugins/classic-editor/) plugin
2. Open any post or page in Classic Editor
3. In the admin bar, click on WP Rocket → Clear and preload cache

**Expected behavior**
User shouldn't be redirected to the post/page list after clearing cache

**Screenshots**

**Additional context**

**Acceptance Criteria (for WP Media team use only)**
Clear instructions for developers, to be added before the grooming

### PR DESCRIPTION
**Pull request by Miraeld**
## Description

When you click on Clear cache in the admin bar, they get redirected to the list of posts.

Fixes #6301 

## Type of change

- Bug fix (non-breaking change which fixes an issue).

## Is the solution different from the one proposed during the grooming?

No

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [ ] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

File: inc/common/admin-bar.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ function rocket_admin_bar( $wp_admin_bar ) {
 		 *
 		 * @param string $uri Current uri
 		 */
-		$referer = (string) apply_filters( 'rocket_admin_bar_referer', esc_url( $uri ) );
+		$referer = (string) apply_filters( 'rocket_admin_bar_referer', esc_url_raw( $uri ) );
```
Comment (remyperona) [inline inc/common/admin-bar.php]: We should escape after the return value of the filter

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/common/admin-bar.php (hunk 1 above)
Comment (Miraeld) [reply inc/common/admin-bar.php]: I've changed it :)

**Comment by Mai-Saad** [comment]
Related test plan https://wpmediaqa.testrail.io/index.php?/runs/view/793

File: inc/common/admin-bar.php [hunk 2]
```diff
@@ -22,7 +22,7 @@ function rocket_admin_bar( $wp_admin_bar ) {
 		 *
 		 * @param string $uri Current uri
 		 */
-		$referer = (string) apply_filters( 'rocket_admin_bar_referer', esc_url( $uri ) );
+		$referer = (string) esc_url_raw( apply_filters( 'rocket_admin_bar_referer', $uri ) );
```
Comment (remyperona) [inline inc/common/admin-bar.php]:
```suggestion
		$referer = (string) apply_filters( 'rocket_admin_bar_referer', $uri );
		$referer = esc_url_raw( $referer );
```

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Review by CrochetFeve0251 — APPROVED** [review]

**Comment by wordpressfan** [comment]
I did the QA for this PR, it works as expected with the following note that is not related to the PR itself and the same behavior is on develop and trunk:

- With weglot, when visiting the translated page and trying to clear the used CSS from the admin bar, the used CSS is not removed as it's not created for this translated page and only is created for the main page and even the cache is not cleared in this case also.

@wp-media/productrocket with weglot, I think we can't change the page structure by adding a new block or any new visible items on the page, we only translate the content so if this is the case, I think we need to open a new github issue so the cache and used CSS should be tied together so once clearing the cache or used css for a page, we need to remove them with the translated pages.

But the PR itself is working properly, Great job @Miraeld 

[testrail-report-551.pdf](https://github.com/wp-media/wp-rocket/files/13684875/testrail-report-551.pdf)

**Pull request merged**